{"uid":"d932418e96584792","name":"Connect new exchange","fullName":"tests.credentials.test_connect_exchange.TestConnectExchange#test_connect_exchange","historyId":"2166a9441b35c6149e02be72864b8f9c","time":{"start":1726944992780,"stop":1726944994986,"duration":2206},"descriptionHtml":"<h4 style=\"margin: -10px 0 5px;\"><span style=\"margin: 9px 4px 9px 0; cursor: pointer\" class=\"fa fa-files-o\" data-copy=\"test_connect_exchange[Poloniex]\"></span>test_connect_exchange[Poloniex]</h4>\n    Collected at <span class=\"label label_status_passed\">2024-09-21 18:54</span><br><br>","status":"broken","statusMessage":"requests.exceptions.HTTPError: 401 Client Error: Unauthorized for url: https://dev.api.darkbot.io/auth/login","statusTrace":"self = <credentials.test_connect_exchange.TestConnectExchange object at 0x7f726fcca5d0>\npage = <Page url='https://darkbot:98713456@dev.app.darkbot.io/'>\ncreate_session = <function create_session.<locals>._create_session at 0x7f726d019080>\nexchange = 'Poloniex'\n\n    @allure.title(\"Connect new exchange\")\n    @pytest.mark.parametrize(**CredentialsParams.ddt_exchange_value)\n    def test_connect_exchange(self, page: Page, create_session, exchange):\n        login_page = LoginPage(page)\n        credentials_page = CredentialsPage(page)\n        notifications = Notifications(page)\n        api_key = {\n            \"Binance\": cfg.BINANCE_API_KEY,\n            \"Poloniex\": cfg.POLONIEX_API_KEY,\n            \"HitBTC\": cfg.HITBTC_API_KEY,\n        }\n        secret_key = {\n            \"Binance\": cfg.BINANCE_SECRET_KEY,\n            \"Poloniex\": cfg.POLONIEX_SECRET_KEY,\n            \"HitBTC\": cfg.HITBTC_SECRET_KEY,\n        }\n    \n        toast_message = \"Credentials was successfully created.\"\n    \n        with allure.step(\"Test preparation\"):\n            login_page.goto()\n>           create_session()\n\napi_key    = {'Binance': 'AJWMNNtmbuIrpuebS2MgPFKBlK5iRTc6zECqk4pw9RjhBLTSzsofpnqGgwOMNwSH', 'HitBTC': 'Mhc4KBMfGmi11WkiqGsAS9LS3M2Dhn2d', 'Poloniex': '9XRJDDFA-X83E7C3T-KVKEXOXI-N65DV9OU'}\ncreate_session = <function create_session.<locals>._create_session at 0x7f726d019080>\ncredentials_page = <src.pages.credentials_page.credentials_page.CredentialsPage object at 0x7f726d4ec350>\nexchange   = 'Poloniex'\nlogin_page = <src.pages.login_page.login_page.LoginPage object at 0x7f726d4ec2c0>\nnotifications = <src.pages.notifications.Notifications object at 0x7f726d4ecf50>\npage       = <Page url='https://darkbot:98713456@dev.app.darkbot.io/'>\nsecret_key = {'Binance': 'Hh4twEoXrdvUvv0q9gAJuhne1WdUMtx9Q2bVfYiT3I4qIdZ8SUVirZmM010sFSLX', 'HitBTC': 'awLcD9SCYFyR8dJYyRor5SEdJLE...783da8515f572cace50d976b8acff04e2dec84541fc2d18b4d4cbfc49ab72c78954bb15146e3bddf3025c4a0f29e8ba3488b231ce8446d4f9b712'}\nself       = <credentials.test_connect_exchange.TestConnectExchange object at 0x7f726fcca5d0>\ntoast_message = 'Credentials was successfully created.'\n\ntests/credentials/test_connect_exchange.py:44: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \nconftest.py:132: in _create_session\n    localStorage.setItem('token', '{prepare_user_session(auth_data=auth_data)}');\n        auth_data  = {'password': 'darkbot123', 'token': 'token', 'username': 'premium@darkbot.io'}\n        page       = <Page url='https://darkbot:98713456@dev.app.darkbot.io/'>\n        prepare_user_session = <function prepare_user_session at 0x7f726cdf65c0>\nmodules/Utils.py:56: in prepare_user_session\n    res = AuthApi(auth_data=auth_data)\n        auth_data  = {'password': 'darkbot123', 'token': 'token', 'username': 'premium@darkbot.io'}\nsrc/api/base_api.py:22: in __init__\n    self._login(auth_data)\n        auth_data  = {'password': 'darkbot123', 'token': 'token', 'username': 'premium@darkbot.io'}\n        jwt_token  = None\n        lang       = 'en'\n        self       = <src.api.auth.register_api.AuthApi object at 0x7f726d49f080>\nsrc/api/base_api.py:30: in _login\n    raise e\n        auth_data  = {'password': 'darkbot123', 'token': 'token', 'username': 'premium@darkbot.io'}\n        response   = <Response [401]>\n        self       = <src.api.auth.register_api.AuthApi object at 0x7f726d49f080>\nsrc/api/base_api.py:27: in _login\n    response.raise_for_status()\n        auth_data  = {'password': 'darkbot123', 'token': 'token', 'username': 'premium@darkbot.io'}\n        response   = <Response [401]>\n        self       = <src.api.auth.register_api.AuthApi object at 0x7f726d49f080>\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nself = <Response [401]>\n\n    def raise_for_status(self):\n        \"\"\"Raises :class:`HTTPError`, if one occurred.\"\"\"\n    \n        http_error_msg = \"\"\n        if isinstance(self.reason, bytes):\n            # We attempt to decode utf-8 first because some servers\n            # choose to localize their reason strings. If the string\n            # isn't utf-8, we fall back to iso-8859-1 for all other\n            # encodings. (See PR #3538)\n            try:\n                reason = self.reason.decode(\"utf-8\")\n            except UnicodeDecodeError:\n                reason = self.reason.decode(\"iso-8859-1\")\n        else:\n            reason = self.reason\n    \n        if 400 <= self.status_code < 500:\n            http_error_msg = (\n                f\"{self.status_code} Client Error: {reason} for url: {self.url}\"\n            )\n    \n        elif 500 <= self.status_code < 600:\n            http_error_msg = (\n                f\"{self.status_code} Server Error: {reason} for url: {self.url}\"\n            )\n    \n        if http_error_msg:\n>           raise HTTPError(http_error_msg, response=self)\nE           requests.exceptions.HTTPError: 401 Client Error: Unauthorized for url: https://dev.api.darkbot.io/auth/login\n\nhttp_error_msg = '401 Client Error: Unauthorized for url: https://dev.api.darkbot.io/auth/login'\nreason     = 'Unauthorized'\nself       = <Response [401]>\n\n/opt/hostedtoolcache/Python/3.12.6/x64/lib/python3.12/site-packages/requests/models.py:1021: HTTPError","flaky":false,"newFailed":false,"newBroken":true,"newPassed":false,"retriesCount":1,"retriesStatusChange":false,"beforeStages":[{"name":"context","time":{"start":1726944992552,"stop":1726944992745,"duration":193},"status":"passed","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":false,"stepsCount":0,"attachmentsCount":0,"hasContent":false},{"name":"page","time":{"start":1726944992746,"stop":1726944992779,"duration":33},"status":"passed","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":false,"stepsCount":0,"attachmentsCount":0,"hasContent":false},{"name":"create_session","time":{"start":1726944992779,"stop":1726944992779,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":false,"stepsCount":0,"attachmentsCount":0,"hasContent":false}],"testStage":{"descriptionHtml":"<h4 style=\"margin: -10px 0 5px;\"><span style=\"margin: 9px 4px 9px 0; cursor: pointer\" class=\"fa fa-files-o\" data-copy=\"test_connect_exchange[Poloniex]\"></span>test_connect_exchange[Poloniex]</h4>\n    Collected at <span class=\"label label_status_passed\">2024-09-21 18:54</span><br><br>","status":"broken","statusMessage":"requests.exceptions.HTTPError: 401 Client Error: Unauthorized for url: https://dev.api.darkbot.io/auth/login","statusTrace":"self = <credentials.test_connect_exchange.TestConnectExchange object at 0x7f726fcca5d0>\npage = <Page url='https://darkbot:98713456@dev.app.darkbot.io/'>\ncreate_session = <function create_session.<locals>._create_session at 0x7f726d019080>\nexchange = 'Poloniex'\n\n    @allure.title(\"Connect new exchange\")\n    @pytest.mark.parametrize(**CredentialsParams.ddt_exchange_value)\n    def test_connect_exchange(self, page: Page, create_session, exchange):\n        login_page = LoginPage(page)\n        credentials_page = CredentialsPage(page)\n        notifications = Notifications(page)\n        api_key = {\n            \"Binance\": cfg.BINANCE_API_KEY,\n            \"Poloniex\": cfg.POLONIEX_API_KEY,\n            \"HitBTC\": cfg.HITBTC_API_KEY,\n        }\n        secret_key = {\n            \"Binance\": cfg.BINANCE_SECRET_KEY,\n            \"Poloniex\": cfg.POLONIEX_SECRET_KEY,\n            \"HitBTC\": cfg.HITBTC_SECRET_KEY,\n        }\n    \n        toast_message = \"Credentials was successfully created.\"\n    \n        with allure.step(\"Test preparation\"):\n            login_page.goto()\n>           create_session()\n\napi_key    = {'Binance': 'AJWMNNtmbuIrpuebS2MgPFKBlK5iRTc6zECqk4pw9RjhBLTSzsofpnqGgwOMNwSH', 'HitBTC': 'Mhc4KBMfGmi11WkiqGsAS9LS3M2Dhn2d', 'Poloniex': '9XRJDDFA-X83E7C3T-KVKEXOXI-N65DV9OU'}\ncreate_session = <function create_session.<locals>._create_session at 0x7f726d019080>\ncredentials_page = <src.pages.credentials_page.credentials_page.CredentialsPage object at 0x7f726d4ec350>\nexchange   = 'Poloniex'\nlogin_page = <src.pages.login_page.login_page.LoginPage object at 0x7f726d4ec2c0>\nnotifications = <src.pages.notifications.Notifications object at 0x7f726d4ecf50>\npage       = <Page url='https://darkbot:98713456@dev.app.darkbot.io/'>\nsecret_key = {'Binance': 'Hh4twEoXrdvUvv0q9gAJuhne1WdUMtx9Q2bVfYiT3I4qIdZ8SUVirZmM010sFSLX', 'HitBTC': 'awLcD9SCYFyR8dJYyRor5SEdJLE...783da8515f572cace50d976b8acff04e2dec84541fc2d18b4d4cbfc49ab72c78954bb15146e3bddf3025c4a0f29e8ba3488b231ce8446d4f9b712'}\nself       = <credentials.test_connect_exchange.TestConnectExchange object at 0x7f726fcca5d0>\ntoast_message = 'Credentials was successfully created.'\n\ntests/credentials/test_connect_exchange.py:44: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \nconftest.py:132: in _create_session\n    localStorage.setItem('token', '{prepare_user_session(auth_data=auth_data)}');\n        auth_data  = {'password': 'darkbot123', 'token': 'token', 'username': 'premium@darkbot.io'}\n        page       = <Page url='https://darkbot:98713456@dev.app.darkbot.io/'>\n        prepare_user_session = <function prepare_user_session at 0x7f726cdf65c0>\nmodules/Utils.py:56: in prepare_user_session\n    res = AuthApi(auth_data=auth_data)\n        auth_data  = {'password': 'darkbot123', 'token': 'token', 'username': 'premium@darkbot.io'}\nsrc/api/base_api.py:22: in __init__\n    self._login(auth_data)\n        auth_data  = {'password': 'darkbot123', 'token': 'token', 'username': 'premium@darkbot.io'}\n        jwt_token  = None\n        lang       = 'en'\n        self       = <src.api.auth.register_api.AuthApi object at 0x7f726d49f080>\nsrc/api/base_api.py:30: in _login\n    raise e\n        auth_data  = {'password': 'darkbot123', 'token': 'token', 'username': 'premium@darkbot.io'}\n        response   = <Response [401]>\n        self       = <src.api.auth.register_api.AuthApi object at 0x7f726d49f080>\nsrc/api/base_api.py:27: in _login\n    response.raise_for_status()\n        auth_data  = {'password': 'darkbot123', 'token': 'token', 'username': 'premium@darkbot.io'}\n        response   = <Response [401]>\n        self       = <src.api.auth.register_api.AuthApi object at 0x7f726d49f080>\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nself = <Response [401]>\n\n    def raise_for_status(self):\n        \"\"\"Raises :class:`HTTPError`, if one occurred.\"\"\"\n    \n        http_error_msg = \"\"\n        if isinstance(self.reason, bytes):\n            # We attempt to decode utf-8 first because some servers\n            # choose to localize their reason strings. If the string\n            # isn't utf-8, we fall back to iso-8859-1 for all other\n            # encodings. (See PR #3538)\n            try:\n                reason = self.reason.decode(\"utf-8\")\n            except UnicodeDecodeError:\n                reason = self.reason.decode(\"iso-8859-1\")\n        else:\n            reason = self.reason\n    \n        if 400 <= self.status_code < 500:\n            http_error_msg = (\n                f\"{self.status_code} Client Error: {reason} for url: {self.url}\"\n            )\n    \n        elif 500 <= self.status_code < 600:\n            http_error_msg = (\n                f\"{self.status_code} Server Error: {reason} for url: {self.url}\"\n            )\n    \n        if http_error_msg:\n>           raise HTTPError(http_error_msg, response=self)\nE           requests.exceptions.HTTPError: 401 Client Error: Unauthorized for url: https://dev.api.darkbot.io/auth/login\n\nhttp_error_msg = '401 Client Error: Unauthorized for url: https://dev.api.darkbot.io/auth/login'\nreason     = 'Unauthorized'\nself       = <Response [401]>\n\n/opt/hostedtoolcache/Python/3.12.6/x64/lib/python3.12/site-packages/requests/models.py:1021: HTTPError","steps":[{"name":"Test preparation","time":{"start":1726944992780,"stop":1726944994979,"duration":2199},"status":"broken","statusMessage":"requests.exceptions.HTTPError: 401 Client Error: Unauthorized for url: https://dev.api.darkbot.io/auth/login\n","statusTrace":"  File \"/home/runner/work/qa-website/qa-website/tests/credentials/test_connect_exchange.py\", line 44, in test_connect_exchange\n    create_session()\n  File \"/home/runner/work/qa-website/qa-website/conftest.py\", line 132, in _create_session\n    localStorage.setItem('token', '{prepare_user_session(auth_data=auth_data)}');\n                                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/home/runner/work/qa-website/qa-website/modules/Utils.py\", line 56, in prepare_user_session\n    res = AuthApi(auth_data=auth_data)\n          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/home/runner/work/qa-website/qa-website/src/api/base_api.py\", line 22, in __init__\n    self._login(auth_data)\n  File \"/home/runner/work/qa-website/qa-website/src/api/base_api.py\", line 30, in _login\n    raise e\n  File \"/home/runner/work/qa-website/qa-website/src/api/base_api.py\", line 27, in _login\n    response.raise_for_status()\n  File \"/opt/hostedtoolcache/Python/3.12.6/x64/lib/python3.12/site-packages/requests/models.py\", line 1021, in raise_for_status\n    raise HTTPError(http_error_msg, response=self)\n","steps":[{"name":"goto route","time":{"start":1726944992780,"stop":1726944994234,"duration":1454},"status":"passed","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":false,"stepsCount":0,"attachmentsCount":0,"hasContent":false}],"attachments":[],"parameters":[],"shouldDisplayMessage":true,"stepsCount":1,"attachmentsCount":0,"hasContent":true}],"attachments":[{"uid":"e35d407dcabaa508","name":"Location","source":"e35d407dcabaa508.uri","type":"text/uri-list","size":44},{"uid":"d7442d1f5f64fa30","name":"Screenshot","source":"d7442d1f5f64fa30.png","type":"image/png","size":175928},{"uid":"12839629e49d1f7d","name":"log","source":"12839629e49d1f7d.txt","type":"text/plain","size":467}],"parameters":[],"shouldDisplayMessage":true,"stepsCount":2,"attachmentsCount":3,"hasContent":true},"afterStages":[{"name":"context::0","time":{"start":1726944995164,"stop":1726944995229,"duration":65},"status":"passed","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":false,"stepsCount":0,"attachmentsCount":0,"hasContent":false}],"labels":[{"name":"suite","value":"Connect exchange"},{"name":"parentSuite","value":"tests.credentials"},{"name":"subSuite","value":"TestConnectExchange"},{"name":"host","value":"fv-az1521-508"},{"name":"thread","value":"5413-MainThread"},{"name":"framework","value":"pytest"},{"name":"language","value":"cpython3"},{"name":"package","value":"tests.credentials.test_connect_exchange"},{"name":"resultFormat","value":"allure2"}],"parameters":[{"name":"exchange","value":"'Poloniex'"}],"links":[],"hidden":false,"retry":false,"extra":{"severity":"normal","retries":[{"uid":"9d5c0502720e5c00","status":"broken","statusDetails":"requests.exceptions.HTTPError: 401 Client Error: Unauthorized for url: https://dev.api.darkbot.io/auth/login","time":{"start":1726944989846,"stop":1726944992154,"duration":2308}}],"categories":[{"name":"Test defects","matchedStatuses":[],"flaky":false}],"history":{"statistic":{"failed":0,"broken":4,"skipped":0,"passed":1,"unknown":0,"total":5},"items":[{"uid":"4d28fd6e87f39819","reportUrl":"https://darkbot-io.github.io/qa-website/531//#testresult/4d28fd6e87f39819","status":"passed","time":{"start":1726943082543,"stop":1726943102397,"duration":19854}},{"uid":"deabe774d87c0951","reportUrl":"https://darkbot-io.github.io/qa-website/529//#testresult/deabe774d87c0951","status":"broken","statusDetails":"playwright._impl._errors.TimeoutError: Page.wait_for_selector: Timeout 30000ms exceeded.\nCall log:\nwaiting for locator(\"//div[@type='success']\") to be visible","time":{"start":1726772453628,"stop":1726772495295,"duration":41667}},{"uid":"2c269732c2539b2","reportUrl":"https://darkbot-io.github.io/qa-website/528//#testresult/2c269732c2539b2","status":"broken","statusDetails":"playwright._impl._errors.TimeoutError: Page.wait_for_selector: Timeout 30000ms exceeded.\nCall log:\nwaiting for locator(\"//div[@type='success']\") to be visible","time":{"start":1726334015087,"stop":1726334055598,"duration":40511}},{"uid":"7a99d8353b4b810d","reportUrl":"https://darkbot-io.github.io/qa-website/527//#testresult/7a99d8353b4b810d","status":"broken","statusDetails":"playwright._impl._errors.TimeoutError: Page.wait_for_selector: Timeout 30000ms exceeded.\nCall log:\nwaiting for locator(\"//div[@type='success']\") to be visible","time":{"start":1725729173758,"stop":1725729215082,"duration":41324}}]},"tags":[]},"source":"d932418e96584792.json","parameterValues":["'Poloniex'"]}